lib/local-cluster/lib/Control/Monitoring/Monitor.hs

```diff
@@ -0,0 +1,151 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TupleSections #-}
+
+-- | A monitor that can switch between pulling and not-pulling
+module Control.Monitoring.Monitor
+    ( MonitorState (..)
+    , Monitor (..)
```

Comment:
```suggestion
    , Monitor (..)
    , tracerFromMonitor
```

Could you add a function

```hs
tracerFromMonitor :: Monitor m a b → Tracer m a
```

here? (Or

```hs
newTracerFromMonitor :: Monitor m a b → m (Tracer m a)
```

if the pure type signature does not work because some resource needs to be allocated.)

The idea is that the `Monitor` implements this fancy functionality about pulling and not pulling, but to the rest of the local cluster and wallet, we simply present it as a `Tracer`.

Reply:
just `monitorTracer` as it's only a field of it